core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1860,6 +1860,30 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+   * Submit a job for execution and return a FutureJob holding the result. Return a Future for
+   * monitoring the job success or failure event.
+   */
+  private[spark] def submitAsyncJob[T, U, R](
```

Comment:
This probably should be marked experimental, since submitJob is experimental.

@JoshRosen Pinging you to review this part of the PR that touches SparkContext.